### ISSUE TITLE
Add Formik validated/submitted with errors handler

### ISSUE THREAD

**Issue by Ronny25** [formium/formik#3048]
### Current Behavior

After validation is done on form submit and we have errors onSubmit callback is never called which is fine.

### Desired Behavior

Formik should have a specific callback handler for this scenario.

### Suggested Solution

There would be really nice to have a callback for exactly this scenario: Form has errors, onSubmit is not called but a specific callback is triggered (if provided).
This will allow handling error case in a more user-friendly way. Let's say to display a very general error/info message for the user that might say: Please check your details. And possibly scroll to it as form might be pretty big, especially on mobile devices. And displaying only field error message is not enough as it's out of the viewport.

### Who does this impact? Who is this for?

This would be nice to have for all the user: developers and end customers.

### Describe alternatives you've considered

So far to achieve the same behaviour I'm checking for 3 things: isSubmitting, isValidating and errors.
Plus, was submission triggered (submit button clicked or Enter key) by passing onClick handler to it.
This allows me to know:
- was submit initiated by the user,
- was validation done,
- was submission done.

If so - I'm adding a general message for the user and scrolling to it.

**Comment by KyorCode** [comment]
Isn't it possible to solve this by using the `errors` object from the formikBag and handle based upon that?

For example:

`
<Formik .....>{({errors}) => {

return (<>
{Object.keys(errors).length > 0 && <ValidationInfo />}
.... other form components
</>)


}}</Formik>
`

**Comment by Ronny25** [comment]
@KyorCode not really as they could appear while typing/changing in a form field.
And I'm asking only after submission. Or at least after trying to submit the form.

**Comment by KyorCode** [comment]
Then throw submitCount in the mix of it. Also available on the formikBag.

**Comment by Ronny25** [comment]
There is kind of a race condition in place.
It seems that submitCount is increased before validation is done.
So I'm having a higher number in submitCount with no error yet 🤷🏻‍♂️

**Comment by KyorCode** [comment]
```javascript`
const ValidationInfo = () => {
const { errors, submitCount } = useFormikContext();

return ({Object.keys(errors}.length > 0 && submitCount > 0 && ( <....> ... </ ... > )})
}
```

It should work:

submitCount = 0 && no errors => no render
submitCount = 1 && errors => render
submitCount = 2 && old Errors = render ( this shouldn't be a problem because the errors will update very soon after, either being no errors, same errors or new errors )
submitCount = 2 && new Errors = render
submitCount = 2 && no errors = no render

**Comment by Ronny25** [comment]
The idea of this Feature request is the following actually:
Formik validates on submission all the fields:
1. form is valid -> calls onSubmit handler;
2. form is not valid -> calls "Another" handler.

I can understand that there are workarounds, I using right now one.
And old errors in your suggestion is actually a problem because scrolling will already be started.
But with such a feature been implemented it would be cleaner and (I hope) useful for many others too.

**Comment by Ronny25** [comment]
Hi @KyorCode ,
so I guess this feature request is completely ignored, right?

**Comment by johnrom** [comment]
@Ronny25 open source moves slowly, and there are no guarantees. if someone wants this feature enough to implement it in a PR, it's possible someone will review it. I think this would be a good example of something we could implement with Plugins #3109

**Comment by Ronny25** [comment]
Hi @johnrom ,
actually, plugin system sounds very good!
Would be awesome to see it implemented in the nearest future and have some examples or good docs how to use it.
But I guess it should be able to solve my issue.
Thanks very much for your reply.
Looking forward to it.
(then I will close this request)
